### ISSUE TITLE
need python 3 support for os_firewall

### ISSUE THREAD

**Issue by dustymabe** [openshift/openshift-ansible#2808]
need python 3 support for os_firewall 
```
TASK [os_firewall : Add iptables allow rules] **********************************
failed: [ec2-54-173-118-171.compute-1.amazonaws.com] (item={u'port': u'2379/tcp', u'service': u'etcd'}) => {"failed": true, "item": {"port": "2379/tcp", "service": "etcd"}, "module_stderr": "Shared connection to ec2-54-173-118-171.compute-1.amazonaws.com closed.\r\n", "module_stdout": "iptables: Bad rule (does a matching rule exist in that chain?).\r\niptables: No chain/target/match by that name.\r\niptables: No chain/target/match by that name.\r\nTraceback (most recent call last):\r\n  File \"/tmp/ansible_ye3skiud/ansible_module_os_firewall_manage_iptables.py\", line 273, in <module>\r\n    main()\r\n  File \"/tmp/ansible_ye3skiud/ansible_module_os_firewall_manage_iptables.py\", line 257, in main\r\n    iptables_manager.add_rule(port, protocol)\r\n  File \"/tmp/ansible_ye3skiud/ansible_module_os_firewall_manage_iptables.py\", line 87, in add_rule\r\n    self.verify_chain()\r\n  File \"/tmp/ansible_ye3skiud/ansible_module_os_firewall_manage_iptables.py\", line 82, in verify_chain\r\n    self.create_jump()\r\n  File \"/tmp/ansible_ye3skiud/ansible_module_os_firewall_manage_iptables.py\", line 142, in create_jump\r\n    input_rules = [s.split() for s in output.split('\\n')]\r\nTypeError: a bytes-like object is required, not 'str'\r\n", "msg": "MODULE FAILURE"}
failed: [ec2-54-173-118-171.compute-1.amazonaws.com] (item={u'port': u'2380/tcp', u'service': u'etcd peering'}) => {"failed": true, "item": {"port": "2380/tcp", "service": "etcd peering"}, "module_stderr": "Shared connection to ec2-54-173-118-171.compute-1.amazonaws.com closed.\r\n", "module_stdout": "iptables: Bad rule (does a matching rule exist in that chain?).\r\niptables: No chain/target/match by that name.\r\nTraceback (most recent call last):\r\n  File \"/tmp/ansible_7zcl14sm/ansible_module_os_firewall_manage_iptables.py\", line 273, in <module>\r\n    main()\r\n  File \"/tmp/ansible_7zcl14sm/ansible_module_os_firewall_manage_iptables.py\", line 257, in main\r\n    iptables_manager.add_rule(port, protocol)\r\n  File \"/tmp/ansible_7zcl14sm/ansible_module_os_firewall_manage_iptables.py\", line 87, in add_rule\r\n    self.verify_chain()\r\n  File \"/tmp/ansible_7zcl14sm/ansible_module_os_firewall_manage_iptables.py\", line 82, in verify_chain\r\n    self.create_jump()\r\n  File \"/tmp/ansible_7zcl14sm/ansible_module_os_firewall_manage_iptables.py\", line 142, in create_jump\r\n    input_rules = [s.split() for s in output.split('\\n')]\r\nTypeError: a bytes-like object is required, not 'str'\r\nChain OS_FIREWALL_ALLOW (0 references)\r\ntarget     prot opt source               destination         \r\n", "msg": "MODULE FAILURE"}
```

##### Version
```
$ git describe
openshift-ansible-3.4.17-1-93-gb6fa0f8
```
```
$ ansible --version
ansible 2.2.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
[vagrant@f25vanilla openshift-ansible]$ rpm -q ansible
ansible-2.2.0.0-3.fc25.noarch
```

##### Steps To Reproduce
1. start an f25 atomic host in aws (`ami-c4052cd3` "pre-release version")
2. run `ansible-playbook -i myinventory playbooks/byo/config.yml -e 'ansible_python_interpreter=/usr/bin/python3' -v` with the following inventory file (you must update variables):

```
# Create an OSEv3 group that contains the masters and nodes groups
[OSEv3:children]
masters
nodes
etcd

# Set variables common for all OSEv3 hosts
[OSEv3:vars]
ansible_user=fedora
ansible_become=true
deployment_type=origin
containerized=true
openshift_release=v1.3.1

# uncomment the following to enable htpasswd authentication; defaults to DenyAllPasswordIdentityProvider
#openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]

# host group for masters
[masters]
ec2-54-173-118-171.compute-1.amazonaws.com openshift_public_hostname=ec2-54-173-118-171.compute-1.amazonaws.com openshift_hostname=10.0.7.191.xip.io

# host group for nodes
[nodes]
ec2-54-173-118-171.compute-1.amazonaws.com openshift_public_hostname=ec2-54-173-118-171.compute-1.amazonaws.com openshift_hostname=10.0.7.191.xip.io

# host group for etcd
[etcd]
ec2-54-173-118-171.compute-1.amazonaws.com openshift_public_hostname=ec2-54-173-118-171.compute-1.amazonaws.com openshift_hostname=10.0.7.191.xip.io
```

##### Current Result
Error (see description)
##### Expected Result
No error

**Comment by cgwalters** [comment]
One thing that's related to this: there was discussion of hard requiring firewalld.  If we go that route we'd need to do package layering most likely.

**Comment by detiber** [comment]
@cgwalters hard requiring firewalld for OpenShift or for Atomic Host? We have a PR open to support firewalld by default

**Comment by dustymabe** [comment]
@detiber I guess what he is saying is will we require firewalld? If so then atomic host would not work for now.

**Comment by detiber** [comment]
@dustymabe no we do not require firewalld currently. Longer term we'd like to move that way, but iptables is still the current default and will be supported as long as necessary.

**Comment by dustymabe** [comment]
This looks to be fixed in f709c76 in #2838.
